tests/CloudEventSdkCompliantTest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\CloudFunctions\Tests;
+
+use BadMethodCallException;
+use Google\CloudFunctions\CloudEvent;
+use Google\CloudFunctions\CloudEventSdkCompliant;
+use PHPUnit\Framework\TestCase;
+use DateTimeImmutable;
+use DateTimeInterface;
+
+/**
+ * @group gcf-framework
+ */
+class CloudEventSdkCompliantTest extends TestCase
+{
+    private CloudEvent $cloudevent;
+
+    public function setUp(): void
+    {
+        $this->cloudevent = new CloudEvent(
+            '1413058901901494',
+            '//pubsub.googleapis.com/projects/MY-PROJECT/topics/MY-TOPIC',
+            '1.0',
+            'com.google.cloud.pubsub.topic.publish',
+            'application/json',
+            'type.googleapis.com/google.logging.v2.LogEntry',
+            'My Subject',
+            '2020-12-08T20:03:19.162Z',
+            [
+                "message" => [
+                    "data" => "SGVsbG8gdGhlcmU=",
+                    "messageId" => "1408577928008405",
+                    "publishTime" => "2020-08-06T22:31:14.536Z"
+                ],
+                "subscription" => "projects/MY-PROJECT/subscriptions/MY-SUB"
+            ]
+        );
```

Comment:
Rather than using the constructor, which is pretty fragile given the number of arguments (and is not copy-pastable), could we deserialized a JSON payload?

`JsonDeserializer::create()->deserializeStructured($payload);`

Reply:
The CloudEvents SDK can't parse this payload do to a bug. I filed a bug on them: https://github.com/cloudevents/sdk-php/issues/35

Until they fix it, the best we can do is provide the same interface.

Reply:
Should we add a TODO, or do you think using this constructor is fine?